tutorials/use-cloud-pubsub-cloud-storage-app-engine/index.md

```diff
@@ -0,0 +1,1314 @@
+
+    ```py
+    class ThumbnailReference(ndb.Model):
+        thumbnail_name = ndb.StringProperty()
+        thumbnail_key = ndb.StringProperty()
+        date = ndb.DateTimeProperty(auto_now_add=True)
+        labels = ndb.StringProperty(repeated=True)
+        original_photo = ndb.StringProperty()
+    ```
+
+1. Create a `MainHandler` with a `get` method for getting information from the server and writing it to the home/notification page HTML file. There are no values to pass into the template yet.
+
+    ```py
+    class MainHandler(webapp2.RequestHandler):
+        def get(self):
+          template_values = {}
+          template = jinja_environment.get_template('[NAME OF YOUR HOME PAGE HTML FILE]')
+          self.response.write(template.render(template_values))
+    ```
+
+1. Create a `PhotosHandler` class with a `get` method for getting information from the server and writing it to the photos page HTML file. This should look similar to the `MainHandler`.
+1. Create a `SearchHandler` class with a `get` method for getting information from the server and writing it to the search page HTML file. This should look similar to the `MainHandler` and `PhotosHandler`.
+1. Create a `ReceiveMessage` class with a `post` method for posting information to ther server. This `post` method will receive Cloud Pub/Sub messages and perform necessary logic using the information. Further detail is in the next section.
+
+    ```py
+    class ReceiveMessage(webapp2.RequestHandler):
+        def post(self):
+            # This method will be filled out in the next section.
+    ```
+
+1. Add the following code at the end of your `main.py` file to connect the web page urls with their corresponding classes.
+
+    ```py
+    app = webapp2.WSGIApplication([
+        ('/', MainHandler),
+        ('/photos', PhotosHandler),
+        ('/search', SearchHandler),
+        ('/_ah/push-handlers/receive_message', ReceiveMessage)
+    ], debug=True)
+    ```
+
+### Checkpoint
+
+1. [Run your application locally](https://cloud.google.com/appengine/docs/standard/python/tools/using-local-server) by running the following command from your host directory:
+
+    ```sh
+    dev_appserver.py .
+    ```
+
+    Visit `localhost:8080` in your web browser to view your web application. You should be able to click on the links to  navigate between the pages of your website, which should all be blank except for the navigation links and page titles.
+
+1. [Deploy your application](https://cloud.google.com/appengine/docs/standard/python/getting-started/deploying-the-application) by running:
+
+    ```sh
+    gcloud app deploy
+    ```
+
+    Your web application should be viewable at `http://[PROJECT ID].appspot.com`. You can either navigate there directly through your web browser or launch your browser and view the app by running the command:
+
+    ```sh
+    gcloud app browse
+    ```
+
+## Creating the notifications page
```

Comment:
Kind of odd to have two headings without intervening text. Maybe add an intro sentence for creating the notifications page?